### ISSUE TITLE
Only one expression can be specified in the select list when the subquery is not introduced with EXISTS

### ISSUE THREAD

**Issue by zbrkic** [linq2db/linq2db#3365]
Hi,
When using inner select and OrderBy, generated SQL is wrong. Additional "1" is selected.

### Generated SQL

```sql
DECLARE @take Int -- Int32
SET     @take = 1

SELECT
	[t2].[Name],
	[t2].[is_empty]
FROM
	(
		SELECT
			(
				SELECT TOP (@take)
					[a].[Name],
					1                   -- additional "1" is selected here
				FROM
					[Assignee] [a]
				WHERE
					[x].[Id] = [a].[TaskId]
			) as [Assignee],
			[t1].[Name],
			[t1].[is_empty]
		FROM
			[Task] [x]
				OUTER APPLY (
					SELECT TOP (@take)
						[a].[Name],
						1 as [is_empty]
					FROM
						[Assignee] [a]
					WHERE
						[x].[Id] = [a].[TaskId]
				) [t1]
	) [t2]
ORDER BY
	[t2].[Assignee]
```

### Steps to reproduce

```c#
using LinqToDB;
using LinqToDB.Mapping;
using System.Collections.Generic;
using System.Linq;

namespace Linq2dbTest
{
    public class InnerSelect
    {
        public void Test()
        {
            var builder = MappingSchema.Default.GetFluentMappingBuilder();

            builder.Entity<Task>()
                .Association(x => x.Assignees, x => x.Id, x => x.TaskId, canBeNull: false);

            builder.Entity<Assignee>();

            using (var db = new Db())
            {

                var query = db.Tasks.Select(x => new
                {
                    Assignee = x.Assignees.Select(a => a.Name).FirstOrDefault()
                });

                var orderedQuery = query.OrderBy(x => x.Assignee);

                // Invalid SQL generated: Only one expression can be specified in the select list when the subquery is not introduced with EXISTS.

                // ...
                //SELECT
                //    (
                //        SELECT TOP(@take)
                //            [a].[Name],
                //            1                          <- problem here
                //        FROM
                //            [Assignee][a]
                //        WHERE
                //            [x].[Id] = [a].[TaskId]
                //    ) as [Assignee],
                //    [t1].[Name],
                // ...
            }
        }

        public class Db : LinqToDB.Data.DataConnection
        {
            public Db() : base(ProviderName.SqlServer2017, "data source=localhost;initial catalog=xxx;integrated security=True;MultipleActiveResultSets=True") { }
            public ITable<Task> Tasks => GetTable<Task>();
        }

        public class Task
        {
            public int Id { get; set; }
            public ICollection<Assignee> Assignees { get; set; }
        }

        public class Assignee
        {
            public int Id { get; set; }
            public int TaskId { get; set; }
            public string Name { get; set; }
        }
    }
}
```

### Environment details
linq2db version: *3.5.2*
Database Server: *SQL Server 2019*
Database Provider: *SQL*
Operating system: *W10*
.NET Framework: *4.8*

**Comment by sdanyliv** [comment]
It is as designed. `Assignee.Name` can be null and we cannot decide correctly when record found or we just found `null` value. So translator adds additional flag `is_empty` 1 - in this case we always know when record found.

**Comment by zbrkic** [comment]
Why do you need to know that? Is there some workaround to generate a valid SQL?
Here is what EF6 generates:
```sql
SELECT 
    [Project2].[Id] AS [Id], 
    [Project2].[C1] AS [C1]
    FROM ( SELECT 
        [Extent1].[Id] AS [Id], 
        (SELECT TOP (1) 
            [Extent2].[Name] AS [Name]
            FROM [dbo].[Assignees] AS [Extent2]
            WHERE [Extent1].[Id] = [Extent2].[TaskId]) AS [C1]
        FROM [dbo].[Tasks] AS [Extent1]
    )  AS [Project2]
    ORDER BY [Project2].[C1] ASC
```

**Comment by MaceWindu** [comment]
Fixed in `6.0.0-preview.1`
